Lib/fontTools/pens/basePen.py

```diff
@@ -141,16 +142,52 @@ def addComponent(self, glyphName, transformation):
 		pass
 
 
-class BasePen(AbstractPen):
+class DecomposingPenMixin(LogMixin):
+	""" Pen mixin class implementing a 'addComponent' method that decomposes
+	components (i.e. draws them onto self as simple contours).
+
+	The constructor takes a single 'glyphSet' positional argument, a dict
+	where the glyphs referenced as components are looked up by their name.
+
+	Subclasses can set the 'skipMissingComponents' attribute to True, if they
+	want to ignore the 'KeyError' exception normally raised when a base glyph
+	is missing from the glyphSet, and replace it by a warning message.
+	"""
+
+	skipMissingComponents = False
+
+	def __init__(self, glyphSet):
+		super(DecomposingPenMixin, self).__init__()
```

Comment:
Say I want to make a class that inherits from both `FilterPen` (#918), as it controls another output pen, and from the `DecomposingPenMixin`, because I want it to decompose the components as well.

If I place the `FilterPen` class after the `DecomposingPenMixin` class in the __mro__, the `super()` call in the latter's `__init__` method would in turn try to call the `FilterPen.__init__` method, but this would fail as it's missing the required `outPen` argument.

If I put `FilterPen` before `DecomposingPenMixin`, it would not call the `addComponent` method that decomposes components, but the one from `FilterPen` would be used instead...

The only way I can think of making that work is cheat and set the private attributes myself without calling the superclasses' `__init__`, but this breaks encapsulation:

```python
class DecomposingFilterPen(DecomposingPenMixin, FilterPen):
    def __init__(self, outPen, glyphSet):
        self._outPen = outPen
        self.glyphSet = glyphSet
```

I'm stuck.